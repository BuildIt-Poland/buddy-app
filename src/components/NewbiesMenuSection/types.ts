import Maybe from 'graphql/tsutils/Maybe';
import { Newbie } from 'buddy-app-schema';

export type NewbiesListProps = {
  newbies: Maybe<Newbie>[];
  onSelect: (id: string) => void;
};
