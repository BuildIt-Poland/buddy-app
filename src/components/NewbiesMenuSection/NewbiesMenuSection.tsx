import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import { Box, Typography } from '@material-ui/core';
import Avatar from '../Avatar';
import { NewbiesListProps } from './types';

const NewbiesMenuSection: React.FC<NewbiesListProps> = props => {
  const { newbies } = props;
  return (
    <>
      <Typography component='p' variant='h4'>
        <Box component='strong' fontWeight={'fontWeightBold'}>
          Newbies
        </Box>
      </Typography>
      {newbies &&
        newbies.map((newbie, index) => {
          return (
            <ListItem button key={index}>
              <Typography component='p' variant='h4'>
                {newbie.name}
              </Typography>
              <Avatar type={'small'} imgSrc={newbie.photo} />
            </ListItem>
          );
        })}
    </>
  );
};

export default NewbiesMenuSection;
