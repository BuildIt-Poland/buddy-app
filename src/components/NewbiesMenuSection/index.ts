import NewbiesMenuSection from './NewbiesMenuSection';

export default NewbiesMenuSection;
