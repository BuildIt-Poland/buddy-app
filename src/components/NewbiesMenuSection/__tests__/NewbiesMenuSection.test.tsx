import React from 'react';
import { create } from 'react-test-renderer';

import NewbiesMenuSection from '../NewbiesMenuSection';

jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('../../UserMenuListItem', () => 'UserMenuListItem');

describe('Component - NewbiesMenuSection', () => {
  test('renders correctly', () => {
    const component = create(
      <NewbiesMenuSection
        newbies={[{ name: ' foo', photo: 'url' }]}
        onSelect={() => null}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
