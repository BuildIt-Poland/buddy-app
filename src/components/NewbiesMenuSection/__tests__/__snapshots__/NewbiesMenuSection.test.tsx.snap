// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - NewbiesMenuSection renders correctly 1`] = `
<Box
  className="makeStyles-wrapper-2"
>
  <Typography
    className="makeStyles-title-1"
    component="h3"
    variant="body1"
  >
    <Box
      component="strong"
      fontWeight="fontWeightBold"
    >
      Newbies
    </Box>
  </Typography>
  <UserMenuListItem
    onItemClick={[Function]}
    user={
      Object {
        "name": " foo",
        "photo": "url",
      }
    }
  />
</Box>
`;
